packages/astro/src/vite-plugin-env/index.ts

```diff
@@ -29,7 +33,7 @@ function getPrivateEnv(
 	const privateEnv: Record<string, string> = {};
 	for (const key in fullEnv) {
 		// Ignore public env var
-		if (envPrefixes.every((prefix) => !key.startsWith(prefix))) {
+		if (isValidIdentifierRe.test(key) && envPrefixes.every((prefix) => !key.startsWith(prefix))) {
```

Comment:
Added a valid identifier check because keys could be `ProgramFiles(x64)` which wouldn't work as `import.meta.env.ProgramFiles(x64)`. This worked before because we do plain string replacements, but I don't think people were relying on this to work.